packages/rainbowkit-siwe-next-auth/src/useSiweNextAuth.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AuthenticationConfig,
+  createAuthenticationAdapter,
+} from '@rainbow-me/rainbowkit';
+import { getCsrfToken, signIn, signOut, useSession } from 'next-auth/react';
+import { useMemo } from 'react';
+import { SiweMessage } from 'siwe';
+
+export interface UseSiweNextAuthOptions {
+  statement?: string;
+}
+
+export function useSiweNextAuth({
+  statement = 'Sign in with Ethereum to the app.',
+}: UseSiweNextAuthOptions = {}): AuthenticationConfig<SiweMessage> {
+  const { status } = useSession();
+  const adapter = useMemo(
+    () =>
+      createAuthenticationAdapter({
+        createMessage: ({ address, chainId, nonce }) =>
+          new SiweMessage({
+            address,
+            chainId,
+            domain: window.location.host,
+            nonce,
+            statement,
+            uri: window.location.origin,
+            version: '1',
```

Comment:
is this ok to be static?

Reply:
Related to this comment: https://github.com/rainbow-me/rainbowkit/pull/636#discussion_r943067086 — maybe we need to avoid these issues by forcing users to manage these options themselves? Trying to support a minimal setup, but we obviously want to avoid going too far.

Reply:
Do we default to `'1'` but support overriding?

Reply:
This is now the default value, but can be overridden if needed via the `getSiweMessageOptions` prop.